Client/Frontend/Browser/Playlist/PlaylistSharedFolder.swift

```diff
@@ -0,0 +1,189 @@
+    }
+    return try JSONDecoder().decode(PlaylistSharedFolderModel.self, from: data)
+  }
+  
+  @MainActor
+  public static func createInMemoryStorage(for model: PlaylistSharedFolderModel) async -> PlaylistFolder {
+    await withCheckedContinuation { continuation in
+      // Create a local shared folder
+      PlaylistFolder.addInMemoryFolder(title: model.folderName,
+                                       creatorName: model.creatorName,
+                                       creatorLink: model.creatorLink,
+                                       sharedFolderId: model.folderId) { folder, folderId in
+        // Add the items to the folder
+        PlaylistItem.addInMemoryItems(model.mediaItems, folderUUID: folderId) {
+          // Items were added
+          continuation.resume(returning: folder)
+        }
+      }
+    }
+  }
+  
+  @MainActor
+  public static func saveToDiskStorage(memoryFolder: PlaylistFolder) async -> String {
+    await withCheckedContinuation({ continuation in
+      PlaylistFolder.saveInMemoryFolderToDisk(folder: memoryFolder) { folderId in
+        PlaylistItem.saveInMemoryItemsToDisk(items: Array(memoryFolder.playlistItems ?? []), folderUUID: folderId) {
+          continuation.resume(returning: folderId)
+        }
+      }
+    })
+  }
+  
+  public static func fetchMediaItemInfo(item: PlaylistSharedFolderModel) async -> [PlaylistInfo] {
+    @Sendable @MainActor
+    func fetchTask(item: PlaylistInfo) async -> PlaylistInfo {
+      await withCheckedContinuation { continuation in
+        var webLoader: PlaylistWebLoader?
+        webLoader = PlaylistWebLoader(
+          certStore: nil,
+          handler: { newItem in
+            if let newItem = newItem {
+              PlaylistManager.shared.getAssetDuration(item: newItem) { duration in
+                let item = PlaylistInfo(name: item.name,
+                                   src: newItem.src,
+                                   pageSrc: newItem.pageSrc,
+                                   pageTitle: item.pageTitle,
+                                   mimeType: newItem.mimeType,
+                                   duration: duration ?? newItem.duration,
+                                   detected: newItem.detected,
+                                   dateAdded: newItem.dateAdded,
+                                   tagId: item.tagId)
+                
+                // Destroy the web loader when the callback is complete.
+                webLoader?.removeFromSuperview()
+                webLoader = nil
+                continuation.resume(returning: item)
+              }
+            } else {
+              // Destroy the web loader when the callback is complete.
+              webLoader?.removeFromSuperview()
+              webLoader = nil
+              continuation.resume(returning: item)
+            }
+          }
+        ).then {
+          UIApplication.shared.keyWindow?.insertSubview($0, at: 0)
```

Comment:
This seems like a very weird thing to do in the middle of a fetch method… Again, we shouldn't be reaching into UIApplication like this

Reply:
Must be done. It's the only way the page of this invisible web-view will load.
I don't have an actual window to add all these invisible web-views too.
The same thing has to be done in CarPlay because there is no such thing as a `window` there.

Reply:
Fixed